includes/class-sensei-course.php

```diff
@@ -2518,12 +2593,14 @@ public static function course_archive_sorting( $query ) {
 		}
 
 		/**
-		 * Filter the sensei archive course order by values
+		 * Filter the sensei archive course order by values.
 		 *
 		 * @since 1.9.0
-		 * @param array $options {
-		 *  @type string $option_value
-		 *  @type string $option_string
+		 * @hook sensei_archive_course_order_by_options
+		 *
+		 * @param {array} $options {
```

Comment:
In JSDoc, do we now have to declare `$options` as an `Object` and then do `$options.option_value`? Ditto for the rest.

Reply:
Apparently there's no way to document associative arrays. I took a look at [what 10up is doing](https://10up.github.io/distributor/dt_syncable_taxonomies.html) and they put it in the comment that it's an associative array, so I've followed suit in https://github.com/Automattic/sensei/pull/3564/commits/15ac533243189a4d703ffe404430af76f97563ee.

Reply:
This seems like a pretty big limitation of this. I'd rather not lose all that pretty critical info (for some of them). Did it show up at all using the WordPress' standard for associative arrays?

Reply:
So this works https://github.com/Automattic/sensei/pull/3564/commits/35629b30ce4822874baa32c579df8ebce3fbf8e6

Reply:
I'm more concerned when it is something like `$args` and we've defined what different items are. An example would be https://github.com/Automattic/sensei/blob/2b41aabe4ea1dd33d3b56f848c36cf71b3ef43ba/includes/admin/class-sensei-learners-main.php#L517-L524